shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -469,6 +469,13 @@
 bool FlutterPlatformViewsController::SubmitFrame(GrDirectContext* gr_context,
                                                  std::shared_ptr<IOSContext> ios_context,
                                                  std::unique_ptr<SurfaceFrame> frame) {
+  if (views_to_dispose_.empty() && composition_order_.empty()) {
```

Comment:
this won't fix the issue because `composition_order_` is mutated by `FlutterPlatformViewsController::PrerollCompositeEmbeddedView`,  `PlatformViewLayer::Preroll` (raster task queue), which still leaves the `flutter_view_` race.

Reply:
I see, the thread merger check below is probably better.